src/Libraries/Microsoft.Extensions.AI.Evaluation.Console/Commands/ReportCommand.cs

```diff
@@ -17,17 +20,37 @@ namespace Microsoft.Extensions.AI.Evaluation.Console.Commands;
 internal sealed partial class ReportCommand(ILogger logger)
 {
     internal async Task<int> InvokeAsync(
-        DirectoryInfo storageRootDir,
+        DirectoryInfo? storageRootDir,
+        Uri? endpointUri,
         FileInfo outputFile,
+        bool openReport,
         int lastN,
         Format format,
         CancellationToken cancellationToken = default)
     {
-        string storageRootPath = storageRootDir.FullName;
-        logger.LogInformation("Storage root path: {storageRootPath}", storageRootPath);
+        IResultStore resultStore;
 
-        var results = new List<ScenarioRunResult>();
-        var resultStore = new DiskBasedResultStore(storageRootPath);
+        if (storageRootDir is not null)
+        {
+            string storageRootPath = storageRootDir.FullName;
+            logger.LogInformation("Storage root path: {storageRootPath}", storageRootPath);
+
+            resultStore = new DiskBasedResultStore(storageRootPath);
+        }
+        else if (endpointUri is not null)
+        {
+            logger.LogInformation("Azure Storage endpoint: {endpointUri}", endpointUri);
+
+            var fsClient = new DataLakeDirectoryClient(endpointUri, new DefaultAzureCredential());
+            resultStore = new AzureStorageResultStore(fsClient);
+        }
+        else
+        {
+            logger.LogError("Either --path or --endpoint must be specified");
```

Comment:
Shouldn't this be caught by the validator? If yes, it may be better to throw an exception (InvalidOperationException) since we don't expect this block to be reachable.

Reply:
Fair enough. Updated.

Reply:
Looks like this particular command may have been missed - but the instances in the other commands have been fixed to throw.